gpkit/varkey.py

```diff
@@ -20,34 +20,30 @@ class VarKey(object):
     """
     new_unnamed_id = Count().next
     subscripts = ("models", "idx")
-    eq_ignores = frozenset(["units", "value"])
-    # ignore value in ==. Also skip units, since pints is weird and the unitstr
-    #    will be compared anyway
 
     def __init__(self, name=None, **kwargs):
+        from . import units as ureg  # update in case user disabled units
+        # NOTE: Python arg handling guarantees 'name' won't appear in kwargs
         self.descr = kwargs
-        # Python arg handling guarantees 'name' won't appear in kwargs
         if isinstance(name, VarKey):
             self.descr.update(name.descr)
         else:
             if name is None:
                 name = "\\fbox{%s}" % VarKey.new_unnamed_id()
             self.descr["name"] = str(name)
-
-            # update in case user has disabled units
-            from . import units as ureg
-            if ureg and "units" in self.descr:
-                units = self.descr["units"]
-                if isinstance(units, Strings):
-                    if units == "-":
-                        del self.descr["units"]  # dimensionless
+            if "units" in self.descr:
+                units = self.descr.pop("units")
+                if ureg:
+                    if isinstance(units, Strings):
+                        if units not in ["", "-"]:  # not dimensionless
+                            self.descr["units"] = Quantity(1.0, units)
+                    elif isinstance(units, Quantity):
+                        self.descr["units"] = units
                     else:
-                        self.descr["units"] = Quantity(1.0, units)
-                elif isinstance(units, Quantity):
-                    self.descr["units"] = units
+                        raise ValueError("units must be either a string"
+                                         " or a Quantity from gpkit.units.")
                 else:
-                    raise ValueError("units must be either a string"
-                                     " or a Quantity from gpkit.units.")
+                    self.descr["unitslabel"] = units
```

Comment:
`unitslabel` is only put in descr if units are disabled, to hold the string that was previously (and confusingly) put in `units` where a real unit would go